File: src/main/java/com/big/dreamer/doccentral/document/carsale/model/PersonDetails.java
package com.big.dreamer.doccentral.document.carsale.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.validation.constraints.NotBlank;

public record PersonDetails(
        @JsonProperty("nombre") @NotBlank String givenName,
        @JsonProperty("apellido") @NotBlank String lastName,
        @JsonProperty("departamento") @NotBlank String state,
        @JsonProperty("domicilio") @NotBlank String settlement,
        @JsonProperty("documento") @NotBlank String document,
        @JsonProperty("genero") @NotBlank String gender,
        @JsonProperty("edad") @NotBlank String age,
        @JsonProperty("oficio") @NotBlank String job) {
}

File: src/main/kotlin/com/big/dreamer/doccentral/car/sale/dto/PersonDetails.kt
package com.big.dreamer.doccentral.car.sale.dto

import com.fasterxml.jackson.annotation.JsonAlias

data class PersonDetails(
    @JsonAlias("nombre")
    val givenName: String = "",
    @JsonAlias("apellido")
    val lastName: String = "",
    @JsonAlias("departamento")
    val state: String = "",
    @JsonAlias("domicilio")
    val settlement: String = "",
    @JsonAlias("documento")
    val document: String = "",
    val nit: String = "",
    @JsonAlias("genero")
    var gender: String = "",
    @JsonAlias("edad")
    val age: String = "",
    @JsonAlias("oficio")
    val job: String = "")
